lib/bitcoin/lock/LockMonitor.ts

```diff
@@ -31,17 +32,21 @@ interface LockState {
  * Encapsulates functionality to monitor and create/remove amount locks on bitcoin.
  */
 export default class LockMonitor {
-  private initialized: boolean;
-
   private periodicPollTimeoutId: NodeJS.Timeout | undefined;
 
   private currentLockState: LockState;

```

Comment:
I think there can be a further simplification here: that we remove this in-memory `currentLockState` completely. Let me know if you agree then I will remove it.

The way I see it:
1. Polling happens only once a few minutes, it's okay if it does a full `getCurrentValueTimeLock()` call.
2. If REST API call occurs for getting the active lock, it is even more okay that we do a full `getCurrentValueTimeLock()` call also because a) the usage pattern should be that this occurs once every 10 minutes or so and b) this ensures that we are getting the latest state of the lock, not some "cached state".

Reply:
The only concern that I have w/ the removal of this is that now an external call to the `getCurrentValueTimeLock()` function can cause reads/re-broadcasts internally. I haven't formed an opinion on that because the usage pattern of the external REST API may prevent that but something to think about.

Reply:
Good point and related to the other comment, I think we address your concern by decouple the rebroadcast by pulling it out from `getCurrentValueTimeLock()`.